app/src/main/java/io/gnosis/safe/ui/splash/SplashActivity.kt

```diff
@@ -69,8 +58,7 @@ class SplashActivity : BaseActivity() {
         )
 
         lifecycleScope.launch {
-            notificationRepo.checkPermissions()
-            notificationRepo.register()
+            viewModel.onAppStart()
```

Comment:
Just a question no blocker (as the behaviour is not changed): If I have a bad internet connection, is the splash screen then shown longer, as we wait for the app to register for push on our backend service?

Reply:
Ohh wait, so after 500ms (or when the user agrees to the terms, we continue) ... doesn't that then kill the lifecycleScope? ... anyways I don'T think it is a big issue as the request is already fired and even if the lifecycleScope is cancelled it should not be able to cancel the processing of the request in the backend